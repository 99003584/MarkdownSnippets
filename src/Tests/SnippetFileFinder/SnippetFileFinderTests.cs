using System.Collections.Concurrent;
using System.IO;
using System.Linq;
using MarkdownSnippets;
using Xunit;
using Xunit.Abstractions;

public class SnippetFileFinderTests :
    XunitApprovalBase
{
    [Fact]
    public void Nested()
    {
        var directory = Path.Combine(AssemblyLocation.CurrentDirectory, "SnippetFileFinder/Nested");
        var extractor = new SnippetFileFinder();
        var files = extractor.FindFiles(directory);
        ObjectApprover.Verify(files, Scrubber.Scrub);
    }

    [Fact]
    public void Simple()
    {
        var directory = Path.Combine(AssemblyLocation.CurrentDirectory, "SnippetFileFinder/Simple");
        var extractor = new SnippetFileFinder();
        var files = extractor.FindFiles(directory);
        ObjectApprover.Verify(files, Scrubber.Scrub);
    }

    [Fact]
    public void VerifyLambdasAreCalled()
    {
        var directories = new ConcurrentBag<string>();
        var targetDirectory = Path.Combine(AssemblyLocation.CurrentDirectory,
            "SnippetFileFinder/VerifyLambdasAreCalled");
        var extractor = new SnippetFileFinder(
            directoryFilter: path =>
            {
                directories.Add(path);
                return true;
            }
        );
        extractor.FindFiles(targetDirectory);
        ObjectApprover.Verify(directories.OrderBy(file => file), Scrubber.Scrub);
    }

    public SnippetFileFinderTests(ITestOutputHelper output) :
        base(output)
    {
    }
}