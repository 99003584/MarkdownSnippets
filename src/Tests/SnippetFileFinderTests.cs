using System.Collections.Concurrent;
using System.IO;
using System.Linq;
using System.Threading.Tasks;
using MarkdownSnippets;
using VerifyXunit;
using Xunit;

[UsesVerify]
public class SnippetFileFinderTests
{
    [Fact]
    public Task Nested()
    {
        var directory = Path.Combine(AssemblyLocation.CurrentDirectory, "SnippetFileFinder/Nested");
        SnippetFileFinder finder = new(_=> true);
        var files = finder.FindFiles(directory);
        return Verifier.Verify(files);
    }

    [Fact]
    public Task Simple()
    {
        var directory = Path.Combine(AssemblyLocation.CurrentDirectory, "SnippetFileFinder/Simple");
        SnippetFileFinder finder = new(_=> true);
        var files = finder.FindFiles(directory);
        return Verifier.Verify(files);
    }

    [Fact]
    public Task VerifyLambdasAreCalled()
    {
        ConcurrentBag<string> directories = new();
        var targetDirectory = Path.Combine(AssemblyLocation.CurrentDirectory,
            "SnippetFileFinder/VerifyLambdasAreCalled");
        SnippetFileFinder finder = new(
            shouldIncludeDirectory: path =>
            {
                directories.Add(path);
                return true;
            }
        );
        finder.FindFiles(targetDirectory);
        return Verifier.Verify(directories.OrderBy(file => file));
    }
}