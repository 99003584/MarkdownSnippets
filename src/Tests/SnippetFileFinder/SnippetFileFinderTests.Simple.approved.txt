[
  'root\\SnippetFileFinder\Simple\code1.txt',
  'root\\SnippetFileFinder\Simple\code2.txt',
  'root\\SnippetFileFinder\Simple\code3.txt',
  'root\\SnippetFileFinder\Simple\code4.txt'
]